released/usecases/services_apps/sap_build_code/step2/main.tf

```diff
@@ -0,0 +1,269 @@
+# ------------------------------------------------------------------------------------------------------
+# Setup sdm
+# ------------------------------------------------------------------------------------------------------
+# Entitle 
+resource "btp_subaccount_entitlement" "sdm" {
+  subaccount_id = var.subaccount_id
+  service_name  = "sdm"
+  plan_name     = "build-code"
+}
+
+# Create the service instance
+data "cloudfoundry_service" "sdm" {
+  name       = "sdm"
+  depends_on = [btp_subaccount_entitlement.sdm]
+}
+resource "cloudfoundry_service_instance" "sdm" {
+  name         = "default_sdm"
+  space        = cloudfoundry_space.dev.id
+  type         = "managed"
+  service_plan = data.cloudfoundry_service.sdm.service_plans["build-code"]
+  depends_on   = [cloudfoundry_space_role.space_manager, cloudfoundry_space_role.space_developer, cloudfoundry_org_role.organization_manager, btp_subaccount_entitlement.sdm]
+}
+# Create service key
+resource "random_id" "service_key_sdm" {
+  byte_length = 12
+}
+resource "cloudfoundry_service_credential_binding" "sdm" {
+  type             = "key"
+  name             = join("_", ["defaultKey", random_id.service_key_sdm.hex])
+  service_instance = cloudfoundry_service_instance.sdm.id
+}
+
+
+# ------------------------------------------------------------------------------------------------------
+# Setup mobile-services
+# ------------------------------------------------------------------------------------------------------
+# Entitle 
+resource "btp_subaccount_entitlement" "mobile_services" {
+  subaccount_id = var.subaccount_id
+  service_name  = "mobile-services"
+  plan_name     = "build-code"
+}
+# Create the service instance
+data "cloudfoundry_service" "mobile_services" {
+  name       = "mobile-services"
+  depends_on = [btp_subaccount_entitlement.mobile_services]
+}
+resource "cloudfoundry_service_instance" "mobile_services" {
+  name         = "default_mobile-services"
+  space        = cloudfoundry_space.dev.id
+  type         = "managed"
+  service_plan = data.cloudfoundry_service.mobile_services.service_plans["build-code"]
+  depends_on   = [cloudfoundry_space_role.space_manager, cloudfoundry_space_role.space_developer, cloudfoundry_org_role.organization_manager, btp_subaccount_entitlement.mobile_services]
+}
+# Create service key
+resource "random_id" "service_key_mobile_services" {
+  byte_length = 12
+}
+resource "cloudfoundry_service_credential_binding" "mobile_services" {
+  type             = "key"
+  name             = join("_", ["defaultKey", random_id.service_key_mobile_services.hex])
+  service_instance = cloudfoundry_service_instance.mobile_services.id
+}
+
+# ------------------------------------------------------------------------------------------------------
+# Setup cloud-logging
+# ------------------------------------------------------------------------------------------------------
+# Entitle 
+resource "btp_subaccount_entitlement" "cloud_logging" {
```

Comment:
See comment form entitlement above -> I would entitle everything in step 1 as this reduces the risk that entitlements are not propagated to the CF Org

Reply:
Revised: To keep things self-contained, we leave the stuff in the CF step